File: src/main/java/guru/xgm/vjass/openapi/fileTypes/VjassFileType.java
package guru.xgm.vjass.openapi.fileTypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import guru.xgm.vjass.icons.VjassIcons;
import guru.xgm.vjass.lang.VjassLanguage;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class VjassFileType extends LanguageFileType {

    public static final VjassFileType INSTANCE = new VjassFileType();

    private VjassFileType() {
        super(VjassLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "VJASS";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "VJASS language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "vj";
    }

    @Override
    public @NotNull Icon getIcon() {
        return VjassIcons.FILE;
    }
}

File: src/main/kotlin/raft/war/language/vjass/VjassFileType.kt
package raft.war.language.vjass

import com.intellij.openapi.fileTypes.LanguageFileType
import raft.war.language.vjass.icons.VjassIcons
import raft.war.language.vjass.lang.VjassLanguage
import javax.swing.Icon

class VjassFileType private constructor() : LanguageFileType(VjassLanguage.instance) {
    override fun getName(): String = VjassLanguage.NAME
    override fun getDescription(): String = VjassLanguage.NAME
    override fun getDefaultExtension(): String = EXTENSION
    override fun getIcon(): Icon = VjassIcons.FILE

    @Suppress("CompanionObjectInExtension")
    companion object {
        val instance: VjassFileType = VjassFileType()
        const val EXTENSION = "vj"
    }
}
